### ISSUE TITLE
An exception in factory `Init()` should cause termination with an error exit code

### ISSUE THREAD

**Issue by veprbl** [eic/EICrecon#625]
### Environment: (where does this bug occur, have you tried other environments)
- Which branch (often `main` for latest released):
- Which version (or `HEAD` for the most recent on git):
- Any specific OS or system where the issue occurs?
- Any special versions of ROOT or Geant4?

### Steps to reproduce: (give a step by step account of how to trigger the bug)
1.
2.

### Expected Result: (what do you expect when you execute the steps above)
Reconstruction terminates

### Actual Result: (what do you get when you execute the steps above)
```
[2023-04-21 18:24:45.464] [JEventProcessorPODIO] [error] Omitting PODIO collection 'FooBarFactory' due to exception: Unknown exception in JFactoryT::Init().
```
and reconstruction continues

**Comment by nathanwbrei** [comment]
I fully agree that we should remove the try/catch block inside JEventProcessorPODIO and let crashing factories crash rather than attempting to keep going. Most of the factories are crashing because there is a JEvent::Get<T>() for some T that is not present in the input file (due to geometry, etc). We can use the non-excepting forms of Get/GetSingle, or we can wrap them in the try/catch block instead.
